src/Adapter/ContextStateManager.php

```diff
@@ -157,6 +157,14 @@ public function setCustomer(?Customer $customer): self
         return $this;
     }
 
+    public function setTranslatorLocale(string $locale): self
```

Comment:
As we discussed this probably can be done via the `setLanguage` method, so that it correctly switched the translator when language is overridden

And it also re-switch back the translator's locale correctly when the state is popped